File: app/src/main/java/com/tutv/android/datasource/retrofit/endpoint/UserAPI.java
package com.tutv.android.datasource.retrofit.endpoint;

import com.tutv.android.datasource.retrofit.annotation.AuthenticatedRequest;
import com.tutv.android.datasource.retrofit.annotation.AvoidForceShowLoginScreenOnAuthFail;
import com.tutv.android.domain.User;

import io.reactivex.Single;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Header;

public interface UserAPI {
    @AvoidForceShowLoginScreenOnAuthFail
    @GET("user")
    Single<User> getCurrentUser(@Header("Authorization") String authHeader);

    @AuthenticatedRequest
    @GET("user")
    Single<User> getCurrentUser();

    @AvoidForceShowLoginScreenOnAuthFail
    @AuthenticatedRequest
    @GET("user")
    Single<User> getCurrentUserNoForceLoginScreenOnFail();
}

File: app/src/main/java/com/tutv/android/datasource/retrofit/endpoint/UserAPI.kt
package com.tutv.android.datasource.retrofit.endpoint

import com.tutv.android.datasource.retrofit.annotation.AuthenticatedRequest
import com.tutv.android.domain.Genre
import com.tutv.android.domain.Series
import com.tutv.android.datasource.dto.ResourceViewedDTO
import com.tutv.android.domain.Episode
import com.tutv.android.datasource.dto.SeriesFollowedDTO
import com.tutv.android.datasource.dto.SeriesFollowedResponseDTO
import com.tutv.android.datasource.retrofit.annotation.AvoidForceShowLoginScreenOnAuthFail
import com.tutv.android.domain.User
import io.reactivex.Single
import retrofit2.http.*

interface UserAPI {
    @AvoidForceShowLoginScreenOnAuthFail
    @GET("user")
    fun getCurrentUser(@Header("Authorization") authHeader: String?): Single<User>

    @get:GET("user")
    @get:AuthenticatedRequest
    val currentUser: Single<User>

    @get:GET("user")
    @get:AuthenticatedRequest
    @get:AvoidForceShowLoginScreenOnAuthFail
    val currentUserNoForceLoginScreenOnFail: Single<User>
}